### PR TITLE
Fix frozen String issue in rails 5 when using "1" as CC #1421

### LINKED ISSUE #1421: RuntimeError in Spree::CheckoutController#update (can't modify frozen String)
Setup: Solidus v2.0.0.rc1
Ruby: 2.3.1 MRI

To reproduce, attempt to purchase an item using the bogus gateway, credit card number of 1 in a standard Solidus sandbox.

# Stacktrace
```
vendor/bundle/gems/solidus_core-2.0.0.rc1/app/models/spree/credit_card.rb:92:in `gsub!'
vendor/bundle/gems/solidus_core-2.0.0.rc1/app/models/spree/credit_card.rb:92:in `set_last_digits'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:382:in `block in make_lambda'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:169:in `block (2 levels) in halting'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:770:in `block (2 levels) in deprecated_false_terminator'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:769:in `catch'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:769:in `block in deprecated_false_terminator'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:170:in `block in halting'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:454:in `block in call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:454:in `each'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:454:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:750:in `_run_save_callbacks'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/callbacks.rb:298:in `create_or_update'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/persistence.rb:125:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/validations.rb:44:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/attribute_methods/dirty.rb:22:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:319:in `block (2 levels) in save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:395:in `block in with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:230:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:211:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:392:in `with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:319:in `block in save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:334:in `rollback_active_record_state!'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:318:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/suppressor.rb:41:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/persistence.rb:266:in `block in update'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:395:in `block in with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:230:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:211:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:392:in `with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/persistence.rb:264:in `update'
vendor/bundle/gems/solidus_core-2.0.0.rc1/app/models/spree/gateway/bogus.rb:20:in `create_profile'
vendor/bundle/gems/solidus_core-2.0.0.rc1/app/models/spree/payment.rb:244:in `create_payment_profile'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:382:in `block in make_lambda'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:207:in `block in halting_and_conditional'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:456:in `block in call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:456:in `each'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:456:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:750:in `_run_save_callbacks'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/callbacks.rb:298:in `create_or_update'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/persistence.rb:125:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/validations.rb:44:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/attribute_methods/dirty.rb:22:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:319:in `block (2 levels) in save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:395:in `block in with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:230:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:211:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:392:in `with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:319:in `block in save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:334:in `rollback_active_record_state!'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:318:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/suppressor.rb:41:in `save'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:501:in `block in save'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:520:in `block in around_save'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:150:in `block in run_actions'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:170:in `catch_exceptions'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:148:in `run_actions'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:133:in `run_callbacks'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:212:in `run_callbacks'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:63:in `block (2 levels) in perform'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:63:in `catch'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:63:in `block in perform'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:186:in `within_transaction'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:62:in `perform'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:520:in `around_save'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:501:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/associations/has_many_association.rb:48:in `insert_record'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/autosave_association.rb:398:in `block in save_collection_association'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/autosave_association.rb:391:in `each'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/autosave_association.rb:391:in `save_collection_association'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/autosave_association.rb:185:in `block in add_autosave_association_callbacks'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/autosave_association.rb:158:in `instance_eval'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/autosave_association.rb:158:in `block in define_non_cyclic_method'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:382:in `block in make_lambda'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:207:in `block in halting_and_conditional'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:456:in `block in call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:456:in `each'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:456:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:750:in `_run_update_callbacks'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/callbacks.rb:306:in `_update_record'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/timestamp.rb:81:in `_update_record'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/persistence.rb:534:in `create_or_update'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/callbacks.rb:298:in `block in create_or_update'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:126:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:506:in `block (2 levels) in compile'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:455:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:750:in `_run_save_callbacks'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/callbacks.rb:298:in `create_or_update'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/persistence.rb:125:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/validations.rb:44:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/attribute_methods/dirty.rb:22:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:319:in `block (2 levels) in save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:395:in `block in with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:211:in `transaction'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:392:in `with_transaction_returning_status'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:319:in `block in save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:334:in `rollback_active_record_state!'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/transactions.rb:318:in `save'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/suppressor.rb:41:in `save'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:501:in `block in save'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:520:in `block in around_save'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:150:in `block in run_actions'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:170:in `catch_exceptions'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:148:in `run_actions'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:133:in `run_callbacks'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:212:in `run_callbacks'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:63:in `block (2 levels) in perform'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:63:in `catch'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:63:in `block in perform'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:186:in `within_transaction'
vendor/bundle/gems/state_machines-0.4.0/lib/state_machines/transition_collection.rb:62:in `perform'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:520:in `around_save'
vendor/bundle/gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:501:in `save'
vendor/bundle/gems/solidus_core-2.0.0.rc1/app/models/spree/order_update_attributes.rb:20:in `apply'
/home/gregor/dev/solidus/frontend/app/controllers/spree/checkout_controller.rb:28:in `update'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal/basic_implicit_render.rb:4:in `send_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/abstract_controller/base.rb:188:in `process_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal/rendering.rb:30:in `process_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/abstract_controller/callbacks.rb:20:in `block in process_action'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:126:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:506:in `block (2 levels) in compile'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:455:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:448:in `block (2 levels) in around'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:286:in `block (2 levels) in halting'
/home/gregor/dev/solidus/frontend/app/controllers/spree/store_controller.rb:42:in `block in lock_order'
vendor/bundle/gems/solidus_core-2.0.0.rc1/app/models/spree/order_mutex.rb:27:in `with_lock!'
/home/gregor/dev/solidus/frontend/app/controllers/spree/store_controller.rb:42:in `lock_order'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:382:in `block in make_lambda'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:285:in `block in halting'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:447:in `block in around'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:455:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:750:in `_run_process_action_callbacks'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:90:in `run_callbacks'
vendor/bundle/gems/actionpack-5.0.0.1/lib/abstract_controller/callbacks.rb:19:in `process_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal/rescue.rb:20:in `process_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/notifications.rb:164:in `block in instrument'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/notifications.rb:164:in `instrument'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal/params_wrapper.rb:248:in `process_action'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
vendor/bundle/gems/actionpack-5.0.0.1/lib/abstract_controller/base.rb:126:in `process'
vendor/bundle/gems/actionview-5.0.0.1/lib/action_view/rendering.rb:30:in `process'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal.rb:190:in `dispatch'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_controller/metal.rb:262:in `dispatch'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/routing/route_set.rb:32:in `serve'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/journey/router.rb:39:in `block in serve'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/journey/router.rb:26:in `each'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/journey/router.rb:26:in `serve'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/routing/route_set.rb:725:in `call'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/engine.rb:522:in `call'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/railtie.rb:193:in `public_send'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/railtie.rb:193:in `method_missing'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/routing/mapper.rb:17:in `block in <class:Constraints>'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/routing/mapper.rb:46:in `serve'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/journey/router.rb:39:in `block in serve'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/journey/router.rb:26:in `each'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/journey/router.rb:26:in `serve'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/routing/route_set.rb:725:in `call'
vendor/bundle/gems/versioncake-3.2.0/lib/versioncake/rack/middleware.rb:17:in `call'
vendor/bundle/gems/warden-1.2.6/lib/warden/manager.rb:35:in `block in call'
vendor/bundle/gems/warden-1.2.6/lib/warden/manager.rb:34:in `catch'
vendor/bundle/gems/warden-1.2.6/lib/warden/manager.rb:34:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/etag.rb:25:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/conditional_get.rb:38:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/head.rb:12:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/session/abstract/id.rb:222:in `context'
vendor/bundle/gems/rack-2.0.1/lib/rack/session/abstract/id.rb:216:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/cookies.rb:613:in `call'
vendor/bundle/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:552:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/callbacks.rb:38:in `block in call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:97:in `__run_callbacks__'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:750:in `_run_call_callbacks'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/callbacks.rb:90:in `run_callbacks'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/callbacks.rb:36:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/executor.rb:12:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/debug_exceptions.rb:49:in `call'
vendor/bundle/gems/web-console-3.3.1/lib/web_console/middleware.rb:131:in `call_app'
vendor/bundle/gems/web-console-3.3.1/lib/web_console/middleware.rb:28:in `block in call'
vendor/bundle/gems/web-console-3.3.1/lib/web_console/middleware.rb:18:in `catch'
vendor/bundle/gems/web-console-3.3.1/lib/web_console/middleware.rb:18:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/rack/logger.rb:36:in `call_app'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/rack/logger.rb:24:in `block in call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/tagged_logging.rb:70:in `block in tagged'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/tagged_logging.rb:26:in `tagged'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/tagged_logging.rb:70:in `tagged'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/rack/logger.rb:24:in `call'
vendor/bundle/gems/sprockets-rails-3.2.0/lib/sprockets/rails/quiet_assets.rb:13:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/request_id.rb:24:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/method_override.rb:22:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/runtime.rb:22:in `call'
vendor/bundle/gems/activesupport-5.0.0.1/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/executor.rb:12:in `call'
vendor/bundle/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/static.rb:136:in `call'
vendor/bundle/gems/rack-2.0.1/lib/rack/sendfile.rb:111:in `call'
vendor/bundle/gems/railties-5.0.0.1/lib/rails/engine.rb:522:in `call'
vendor/bundle/gems/puma-3.6.0/lib/puma/configuration.rb:225:in `call'
vendor/bundle/gems/puma-3.6.0/lib/puma/server.rb:578:in `handle_request'
vendor/bundle/gems/puma-3.6.0/lib/puma/server.rb:415:in `process_client'
vendor/bundle/gems/puma-3.6.0/lib/puma/server.rb:275:in `block in run'
vendor/bundle/gems/puma-3.6.0/lib/puma/thread_pool.rb:116:in `block in spawn_thread'
```

### PR DESCRIPTION
**Pull request by jhawthorn**
Fix #1421 and some related cleanup

### CODE REVIEW

**Comment by gmacdougall** [comment]
:+1:

File: core/app/models/spree/credit_card.rb [hunk 1]
```diff
@@ -66,11 +66,10 @@ def expiry=(expiry)
     #
     # @param num [String] the desired credit card number
     def number=(num)
-      @number = begin
-                  num.gsub(/[^0-9]/, '')
-                rescue
-                  nil
-                end
+      @number =
+        if num.is_a?(String)
```
Comment (tvdeyen) [inline core/app/models/spree/credit_card.rb]: It's often better to use duck typing instead of class comparison.

```ruby
if num.respond_to?(:gsub)
```

**Comment by tvdeyen** [comment]
beside my nitpickery LGTM 👍

File: core/app/models/spree/credit_card.rb (hunk 1 above)
Comment (jordan-brough) [inline core/app/models/spree/credit_card.rb]: Do we know what other kinds of values do we expect here, besides nil?  E.g. `Numeric` values?  Is it worth updating the `# @param num [String]` method documentation, or deprecating unexpected value types, or calling `to_s` on everything?  (I don't want to bikeshed so I'm happy for this question to be a separate discussion from this PR)

File: core/app/models/spree/credit_card.rb [hunk 2]
```diff
@@ -91,7 +91,6 @@ def cc_type=(type)
 
     # Sets the last digits field based on the assigned credit card number.
     def set_last_digits
-      number.to_s.gsub!(/\s/, '')
```
Comment (jordan-brough) [inline core/app/models/spree/credit_card.rb]: Does this mean that we no longer automatically strip whitespace from credit card numbers at the model level? Is that worth a changelog entry?

**Comment by jordan-brough** [comment]
👍

File: core/app/models/spree/credit_card.rb (hunk 1 above)
Comment (jhawthorn) [inline core/app/models/spree/credit_card.rb]: Let's discuss this elsewhere. I would like to follow this up and replace this method with something that more resembles a standard AR string attribute (that will cast Integers to a string), but I felt that that was too big of a change here.

**Pull request merged**